src/parquet/ReadParquet.cpp

```diff
@@ -95,6 +95,65 @@ void readColumnIrregularBitWidth(void* chpl_arr, int startIdx, std::shared_ptr<p
   }
 }
 
+int cpp_readStrColumnByName(const char* filename, void* chpl_arr, const char* colname, int64_t batchSize, char** errMsg) {
+  try {
+    int64_t ty = cpp_getType(filename, colname, errMsg);
+  
+    std::unique_ptr<parquet::ParquetFileReader> parquet_reader =
+      parquet::ParquetFileReader::OpenFile(filename, false);
+
+    std::shared_ptr<parquet::FileMetaData> file_metadata = parquet_reader->metadata();
+    int num_row_groups = file_metadata->num_row_groups();
+
+    int64_t i = 0;
+    for (int r = 0; r < num_row_groups; r++) {
+      std::shared_ptr<parquet::RowGroupReader> row_group_reader =
+        parquet_reader->RowGroup(r);
+
+      int64_t values_read = 0;
+
+      std::shared_ptr<parquet::ColumnReader> column_reader;
+
+      auto idx = file_metadata -> schema() -> ColumnIndex(colname);
+      auto max_def = file_metadata -> schema() -> Column(idx) -> max_definition_level(); // needed to determine if nulls are allowed
+
+      if(idx < 0) {
+        std::string dname(colname);
+        std::string fname(filename);
+        std::string msg = "Dataset: " + dname + " does not exist in file: " + fname; 
+        *errMsg = strdup(msg.c_str());
+        return ARROWERROR;
+      }
+      
+      column_reader = row_group_reader->Column(idx);
+
+      if(ty == ARROWSTRING) {
+        int16_t definition_level; // nullable type and only reading single records in batch
+        auto chpl_ptr = (unsigned char*)chpl_arr;
+        parquet::ByteArrayReader* reader =
+          static_cast<parquet::ByteArrayReader*>(column_reader.get());
+
+        while (reader->HasNext()) {
+          parquet::ByteArray value;
+          (void)reader->ReadBatch(1, &definition_level, nullptr, &value, &values_read);
+          // if values_read is 0, that means that it was a null value
+          if(values_read > 0) {
+            for(int j = 0; j < value.len; j++) {
+              chpl_ptr[i] = value.ptr[j];
+              i++;
+            }
+          }
+          i++; // skip one space so the strings are null terminated with a 0
+        }        
+      }
+    }
+    return 0;
+  } catch (const std::exception& e) {
+    *errMsg = strdup(e.what());
+    return ARROWERROR;
+  }
+}
+
 int cpp_readColumnByName(const char* filename, void* chpl_arr, bool* where_null_chpl, const char* colname, int64_t numElems, int64_t startIdx, int64_t batchSize, int64_t byteLength, bool hasNonFloatNulls, char** errMsg) {
```

Comment:
we should be able to remove the `if(ty == ARROWSTRING)` case of `cpp_readColumnByName` now right?

Reply:
I think still want it as an extra line of defense. We get the type just prior to this and if by some error is actually not a string column, then that would crash the code, where here it would throw an error. I don't see much of a downside, so that's why I left it in, even though the upside is probably incredibly unrealistic

Reply:
oh I didn't even think of that, makes sense to me!